templates/indel_classifier.py

```diff
@@ -0,0 +1,268 @@
+
+    # Process deletions: Filter by 'del'/'delin' and handle empty DataFrame
+    dels_df = df[df['Modification'].isin(['del', 'delin'])]
+    if dels_df.empty:
+        grouped_dels = pd.DataFrame(columns=['Start', 'Length', 'Count'])
+    else:
+        grouped_dels = dels_df.groupby(['Start', 'Length']).size().reset_index(name='Count')
+
+    return grouped_dels, grouped_ins
+
+
+def additional_indels_cols(df):
+    # Calculate percentages for in-frame and out-of-frame deletions and insertions
+    # Initialize the columns to store the sums of outframe and inframe deletions and insertions
+    df['Outframe'] = 0
+    df['Inframe'] = 0
+
+    # Check if the necessary base columns exist before attempting calculations
+    required_columns = ['Dels_inframe', 'Dels_outframe', 'Ins_inframe', 'Ins_outframe', 'Total Reads']
+    if all(col in df.columns for col in required_columns):
+        # Aggregate inframe and outframe mutations
+        df['Inframe'] = df['Dels_inframe'] + df['Ins_inframe']
+        df['Outframe'] = df['Dels_outframe'] + df['Ins_outframe']
+
+        # Calculate the percentage for Inframe and Outframe
+        df['% Inframe'] = (df['Inframe'] / df['Total Reads']).fillna(0) * 100
+        df['% Outframe'] = (df['Outframe'] / df['Total Reads']).fillna(0) * 100
+
+        # Handle any potential division by zero issues by replacing infinities with zero
+        df['% Inframe'] = df['% Inframe'].replace([np.inf, -np.inf], 0)
+        df['% Outframe'] = df['% Outframe'].replace([np.inf, -np.inf], 0)
+    else:
+        # If any essential columns are missing, set default percentage values to zero
+        df['% Inframe'] = 0
+        df['% Outframe'] = 0
+
+    # Now, df contains two new columns: '% Inframe' and '% Outframe' with the calculated percentages.
+    return df
+
+
+
+def main():
+
+    min_read_threshold = 200
+
+    if len(sys.argv) < 3:
+        print("Usage: python script.py indels.csv edits.csv")
+        sys.exit(1)
+
+    indel_csv_path = sys.argv[1]
+    edits_csv_path = sys.argv[2]
+
+    grouped_dels, grouped_ins = parse_indels(indel_csv_path)  
+    # Load edits data
+    edits_df = pd.read_csv(edits_csv_path)
+    # Rename the first column which currently has a blank name
+    edits_df.rename(columns={edits_df.columns[0]: 'Sample'}, inplace=True)
+    edits_df = parse_edits_csv(edits_df)
+    edits_df = additional_indels_cols(edits_df)
+    # Initialise zero values in new columns
+    edits_df = edits_df.assign(
+        Class_Conf=0,
+        max_peak=0,
+        av_peak=0,
+        peak_occupancy=0
+    )
+
+    analysis_results = analyze_clonality(grouped_dels, grouped_ins, edits_df, min_read_threshold)
+    # Combine with analysis results
+    for key in analysis_results:
+        edits_df[key] = analysis_results[key]
+
+    outfile = edits_csv_path.replace('.csv','_classified.csv')
+    edits_df.to_csv(outfile)
+    print(edits_df)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
As this will be a template, not a script, we also don't need the `if __name__ == "__main__"` line, we can directly call the function that we want to execute.
```suggestion
main()
```